src/WebJobs.Script.WebHost/Controllers/AdminController.cs

```diff
@@ -48,7 +48,12 @@ public HttpResponseMessage Invoke(string name, [FromBody] FunctionInvocation inv
             {
                 { inputParameter.Name, invocation.Input }
             };
-            Task.Run(() => _scriptHostManager.Instance.CallAsync(function.Name, arguments));
+            var runTask = Task.Run(() => _scriptHostManager.Instance.CallAsync(function.Name, arguments));
+
+            if (block)
+            {
+                await runTask;
```

Comment:
If we're waiting for completion, we don't want to return an Accepted status code, but rather an 200 or 500 based on whether the invocation completed successfully.